src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -2086,103 +2302,60 @@ private bool UseUpdatedEscapeRulesForInvocation(Symbol symbol)
                 argRefKindsOpt,
                 argsToParamsOpt,
-                isRefEscape: false,
-                checkingMethodArgumentsMustMatch: true,
+                includeRefEscape: true,
                 ignoreArglistRefKinds: false,
-                argsAndParamsAll);
-            foreach (var argAndParam in argsAndParamsAll)
-            {
-                var argument = argAndParam.Argument;
-                uint argEscape = argAndParam.IsRefEscape ?
-                    GetRefEscape(argument, scopeOfTheContainingExpression) :
-                    GetValEscape(argument, scopeOfTheContainingExpression);
-
-                if (argEscape > escapeScope)
-                {
-                    escapeScope = argEscape;
-                    argAndParamToEscape = argAndParam;
-                }
-            }
-            argsAndParamsAll.Free();
+                mixableArguments,
+                escapeValues);
 
-            if (argAndParamToEscape is null)
+            var valid = true;
+            foreach (var mixableArg in mixableArguments)
             {
-                return true;
-            }
-
-            var argsAndParams = ArrayBuilder<(ParameterSymbol? Parameter, BoundExpression Argument, RefKind RefKind)>.GetInstance();
-            GetInvocationArgumentsForEscape(
-                symbol,
-                receiverOpt,
-                parameters,
-                argsOpt,
-                argRefKindsOpt: default,
-                argsToParamsOpt,
-                ignoreArglistRefKinds: false,
-                argsAndParams);
-
-            // 2. All `ref` or `out` arguments of `ref struct` types must be assignable by a value with that *safe-to-escape*.
-            //    This applies even when the `ref` argument matches a `scoped ref` parameter.
-            bool result = true;
-            foreach (var (_, argument, refKind) in argsAndParams)
-            {
-                if (!refKind.IsWritableReference() || argument.Type?.IsRefLikeType != true)
+                var toArgEscape = GetValEscape(mixableArg.Argument, scopeOfTheContainingExpression);
+                foreach (var (fromParameter, fromArg, escapeKind, isRefEscape) in escapeValues)
                 {
-                    continue;
-                }
+                    if (mixableArg.Parameter is not null && object.ReferenceEquals(mixableArg.Parameter, fromParameter))
+                    {
+                        continue;
+                    }
 
-                uint escapeTo = GetValEscape(argument, scopeOfTheContainingExpression);
-                var (parameterToEscape, argToEscape, isRefEscape) = argAndParamToEscape.GetValueOrDefault();
+                    // This checks to see if the EscapeValue could ever be assigned to this argument based 
+                    // on comparing the EscapeLevel of both. If this could never be assigned due to 
+                    // this then we don't need to consider it for MAMM analysis
```

Comment:
>analysis [](http://example.com/codeflow?start=71&length=8)

Consider ending multi-sentence comments with ".", for consistency, and to avoid cases where it looks like the last sentence was cut off.